src/raftstore/store/peer.rs

```diff
@@ -275,47 +320,59 @@ impl Peer {
         }))
     }
 
-    pub fn propose_pending_cmd(&mut self, pending_cmd: &mut PendingCmd) -> Result<()> {
-        if pending_cmd.cmd.is_none() {
-            // This may only occur in re-propose.
-            debug!("pending command msg is none for region {} in peer {:?}",
-                   self.region_id,
-                   self.peer);
-            return Ok(());
-        }
-
-        debug!("propose command with uuid {:?}", pending_cmd.uuid);
-
-        // We handle change_peer command as ConfChange entry, and others as normal entry.
-        if let Some(change_peer) = get_change_peer_cmd(pending_cmd.cmd.as_ref().unwrap()) {
-            let data = try!(pending_cmd.cmd.as_ref().unwrap().write_to_bytes());
-
-            let mut cc = raftpb::ConfChange::new();
-            cc.set_change_type(change_peer.get_change_type());
-            cc.set_node_id(change_peer.get_peer().get_id());
-            cc.set_context(data);
-
-            info!("propose conf change {:?} peer {:?} at region {}",
-                  cc.get_change_type(),
-                  cc.get_node_id(),
-                  self.region_id);
-
-            try!(self.raft_group.propose_conf_change(cc));
-            return Ok(());
+    pub fn propose(&mut self, mut cmd: PendingCmd, mut err_resp: RaftCmdResponse) -> Result<()> {
+        if self.pending_cmds.contains(&cmd.uuid) {
+            cmd_resp::bind_error(&mut err_resp, box_err!("duplicated uuid {:?}", cmd.uuid));
+            return cmd.cb.call_box((err_resp,));
+        }
+        debug!("propose command with uuid {:?}", cmd.uuid);
+        if get_change_peer_cmd(cmd.cmd.as_ref().unwrap()).is_some() {
+            if !self.raft_group.raft.pending_conf {
+                // there must be no pending conf changes.
+                assert!(self.pending_cmds.conf_changes.is_empty(),
```

Comment:
if we send multi conf changs, 1, 2 first, when 1 is finished, the raft pend conf is false, but pending cmds conf changes is not empty, but now we send conf change 3.  
can this happen?